sdk/core/Azure.Core.Experimental/src/Serialization/JsonObjectSerializer.cs

```diff
@@ -59,5 +67,85 @@ public override async ValueTask<object> DeserializeAsync(Stream stream, Type ret
         {
             return await JsonSerializer.DeserializeAsync(stream, returnType, _options, cancellationToken).ConfigureAwait(false);
         }
+
+        /// <inheritdoc/>
+        public string? GetSerializedMemberName(MemberInfo memberInfo)
+        {
+            Argument.AssertNotNull(memberInfo, nameof(memberInfo));
+
+            if (!_cache.TryGetValue(memberInfo.ReflectedType, out MemberInfo[] members))
+            {
+                members = GetMembers(memberInfo.ReflectedType).ToArray();
+            }
```

Comment:
It doesn't look like you ever put it in the `_cache`?  We should also make sure access here is thread safe.

Reply:
Good catch. I can lock access, but concurrency is only a concern for mutating the dictionary. Occasionally looping over properties in separate threads isn't really a concern here that should necessitate over locking.

Reply:
Or maybe it doesn't matter since we'll put the same value in and who cares if the last thread wins?